staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -57,12 +71,26 @@ func NewYAMLSerializer(meta MetaFactory, creater runtime.ObjectCreater, typer ru
 	}
 }
 
+// NewStrictYAMLSerializer is the same as NewYAMLSerializer except that it creates a strict YAML serializer
+// that can also return errors of type strictDecoderError.
+// Note that this serializer is not as performant as the non-strict variant, and should not be used in fast paths.
+func NewStrictYAMLSerializer(meta MetaFactory, creater runtime.ObjectCreater, typer runtime.ObjectTyper) *Serializer {
+	return &Serializer{
+		meta:    meta,
+		creater: creater,
+		typer:   typer,
+		yaml:    true,
```

Comment:
Perhaps yaml should be another option in an options struct.

The pattern in this file right now causes us to need 2^N constructors, which I do not think is sensible even if N is only 3. In addition it makes it impossible to layer functions that assemble serializers--we are not going to make 2^N copies of NewUniversalDecoder.

So, I think we should change how things are done here.